dcos/jsonitem.py

```diff
@@ -114,6 +122,22 @@ def clean_value(value):
         return value
 
 
+def _validate_value(key, value, pattern):
+    """
+    :param key: Property key
+    :type key: str
+    :param value: Property value to validate
+    :type value: str
+    :param pattern: Validation pattern
+    :type pattern: str
+    :rtype: None
+    """
+
+    if pattern and not re.match(pattern, value):
```

Comment:
Why Did we have to write our own pattern matching using `import re`? Is it because jsonschema's validation error is not user friendly enough?